python/ray/_private/accelerators/npu.py

```diff
@@ -0,0 +1,103 @@
+import os
+import glob
+import logging
+from typing import Optional, List, Tuple
+
+from ray._private.accelerators.accelerator import AcceleratorManager
+
+logger = logging.getLogger(__name__)
+
+ASCEND_VISIBLE_DEVICES_ENV_VAR = "ASCEND_VISIBLE_DEVICES"
+NOSET_ASCEND_VISIBLE_DEVICES_ENV_VAR = "RAY_EXPERIMENTAL_NOSET_ASCEND_VISIBLE_DEVICES"
+
+
+class NPUAcceleratorManager(AcceleratorManager):
+    """Ascend NPU accelerators."""
+
+    @staticmethod
+    def get_resource_name() -> str:
+        return "NPU"
+
+    @staticmethod
+    def get_visible_accelerator_ids_env_var() -> str:
+        return ASCEND_VISIBLE_DEVICES_ENV_VAR
+
+    @staticmethod
+    def get_current_process_visible_accelerator_ids() -> Optional[List[str]]:
+        ascend_visible_devices = os.environ.get(
+            NPUAcceleratorManager.get_visible_accelerator_ids_env_var(), None
+        )
+
+        if ascend_visible_devices is None:
+            return None
+
+        if ascend_visible_devices == "":
+            return []
+
+        if ascend_visible_devices == "NoDevFiles":
+            return []
+
+        return list(ascend_visible_devices.split(","))
+
+    @staticmethod
+    def get_current_node_num_accelerators() -> int:
+        """Attempt to detect the number of NPUs on this machine.
+
+        NPU chips are represented as devices within `/dev/`, either as `/dev/davinci?`.
+
+        Returns:
+            The number of NPUs if any were detected, otherwise 0.
+        """
+        try:
+            import acl
+
+            device_count, ret = acl.rt.get_device_count()
+            if ret == 0:
+                return device_count
+        except Exception as e:
+            logger.debug("Could not import AscendCL: %s", e)
+
+        try:
+            npu_files = glob.glob("/dev/davinci?")
+            return len(npu_files)
+        except FileNotFoundError as e:
+            logger.debug("Failed to detect number of NPUs: %s", e)
+        return 0
+
+    @staticmethod
+    def get_current_node_accelerator_type() -> Optional[str]:
+        """Get the type of the Ascend NPU on the current node.
+
+        Returns:
+            A string of the type, such as "Ascend910A", "Ascend910B", "Ascend310P1".
+        """
+        try:
+            import acl
+
+            return acl.get_soc_name()
+        except Exception:
+            logger.exception("Failed to detect NPU type.")
+        return None
+
+    @staticmethod
+    def validate_resource_request_quantity(
+        quantity: float,
+    ) -> Tuple[bool, Optional[str]]:
+        if quantity > 1:
+            logger.warning(
+                "The task is requesting multiple Ascend NPUs. "
+                "If you need to build an HCCL network for NPU interconnection, "
+                "please refer to the HCCL User Manual."
```

Comment:
I feel it should be in the documentation not here. Also this applies when I start two actors each with one NPU and they want to talk to each other?

Reply:
The communication between NPUs is similar to the Nvidia's NVLink. If user want to achieve npu2npu communication, they need to use HCCL (similar to NCCL), whether it's within an actor or between actors. It would be appropriate to provide detailed information about this in the documentation. Which document should I add the instructions for using NPU?

Reply:
The doc doesn't exist yet. I'll let you know later and we don't need to do in this PR.